src/machine/machine_avr.go

```diff
@@ -271,3 +272,122 @@ func (i2c I2C) ReadByte() byte {
 
 	return byte(*avr.TWDR)
 }
+
+// UART
+
+type UARTConfig struct {
+	Baudrate uint32
+}
+
+type UART struct {
+}
+
+var (
+	// UART0 is the hardware serial port on the AVR.
+	UART0 = &UART{}
+)
+
+// Configure the UART on the AVR.
+func (uart UART) Configure(config UARTConfig) {
+	// default to 115200
+	if config.Baudrate == 0 {
+		config.Baudrate = 115200
+	}
+
+	// Set baud rate based on prescale formula:
+	// BAUD_PRESCALE = (((F_CPU / (BAUDRATE * 16UL))) - 1)
+	ps := (CPU_FREQUENCY / (config.Baudrate * 16)) - 1
+	*avr.UBRR0H = avr.RegValue(ps >> 8)
+	*avr.UBRR0L = avr.RegValue(ps)
+
+	// enable RX interrupt
+	*avr.UCSR0B |= avr.UCSR0B_RXCIE0
+}
+
+// Close the UART on the AVR.
```

Comment:
If if was a software serial interface, then could be something to close. You are correct, it is just here for the `io.Closer` interface.

Reply:
To be honest, I don't really see the need. It may even be confusing. What does closing a peripheral do? Power it off? Deconfigure it?
If you want to be compatible with Go running under an OS, you will also need to open the device somewhere and closing it is probably done at the same time using `defer`. Code that uses any kind of UART and that should be portable across platforms (desktop and MCU) should not be bothered with opening and closing it as it is platform specific.

For now, I would suggest removing it. It can always be added again if/when the need arises.